Libraries/LibWeb/WebAssembly/Memory.cpp

```diff
@@ -29,17 +37,19 @@ WebIDL::ExceptionOr<GC::Ref<Memory>> Memory::construct_impl(JS::Realm& realm, Me
     if (!address.has_value())
         return vm.throw_completion<JS::TypeError>("Wasm Memory allocation failed"sv);
 
-    auto memory_object = realm.create<Memory>(realm, *address);
+    auto memory_object = realm.create<Memory>(realm, *address, shared);
+
     cache.abstract_machine().store().get(*address)->successful_grow_hook = [memory_object] {
         MUST(memory_object->reset_the_memory_buffer());
     };
 
     return memory_object;
 }
 
-Memory::Memory(JS::Realm& realm, Wasm::MemoryAddress address)
+Memory::Memory(JS::Realm& realm, Wasm::MemoryAddress address, bool shared)
```

Comment:
Please use an enum for "shared" (Shared::No/Shared::Yes) instead.

Reply:
Good idea! Done :)